### ISSUE TITLE
'.bin' directory not available?

### ISSUE THREAD

**Issue by mmorearty** [bazelbuild/rules_nodejs#155]
When you `npm install` a module, if that module's `package.json` includes a `"bin":` attribute, then a `node_modules/.bin/<filename>` file will be created. But although rules_nodejs does seem to create a `.bin/` directory, that directory is not available in all of the places where I think it should be.

To reproduce:

* Download and unzip [example.zip](https://github.com/bazelbuild/rules_nodejs/files/1809665/example.zip)
* `bazel build :main`; then look at `bazel-bin/out.txt`, which contains a list of all of its input files. This include everything from `node_modules/typescript/` (since the "typescript" package is specified in package.json); but it does _not_ include `node_modules/.bin/tsc`. This is surprising, because if you just ran `npm install typescript` without involving Bazel, you would get a `node_modules/.bin/tsc`.
* `bazel build :tsc1`; then `bazel-bin/tsc1`. It works. The `tsc1` target runs `node_modules/typescript/bin/tsc`.
* But the equivalent with `.bin` does not work. `bazel build :tsc2`; then `bazel-bin/tsc2`. You get an error. `tsc2` depends on `node_modules/.bin/tsc`.

**Comment by alexeagle** [comment]
Could you try `bazel query 'deps(@name_of_the_installed_npm_install_repo//:node_modules)'`
to see if the file is being filtered by Bazel's globbing?

**Comment by mmorearty** [comment]
The output of `bazel query 'deps(@foo//:node_modules)` does not include anything from `.bin`. But it DOES include some other filenames that starts with `.`. Here is the output for the above example.zip — as you can see, at the bottom there are three files whose names start with `.`:

```
@foo//:node_modules
@foo//:node_modules/typescript/yarn.lock
@foo//:node_modules/typescript/package.json
@foo//:node_modules/typescript/lib/zh-TW/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/zh-CN/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/watchGuard.js
@foo//:node_modules/typescript/lib/typingsInstaller.js
@foo//:node_modules/typescript/lib/typescriptServices.js
@foo//:node_modules/typescript/lib/typescriptServices.d.ts
@foo//:node_modules/typescript/lib/typescript.js
@foo//:node_modules/typescript/lib/typescript.d.ts
@foo//:node_modules/typescript/lib/tsserverlibrary.js
@foo//:node_modules/typescript/lib/tsserverlibrary.d.ts
@foo//:node_modules/typescript/lib/tsserver.js
@foo//:node_modules/typescript/lib/tsc.js
@foo//:node_modules/typescript/lib/tr/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/ru/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/pt-BR/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/protocol.d.ts
@foo//:node_modules/typescript/lib/pl/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/lib.webworker.d.ts
@foo//:node_modules/typescript/lib/lib.scripthost.d.ts
@foo//:node_modules/typescript/lib/lib.esnext.promise.d.ts
@foo//:node_modules/typescript/lib/lib.esnext.full.d.ts
@foo//:node_modules/typescript/lib/lib.esnext.d.ts
@foo//:node_modules/typescript/lib/lib.esnext.asynciterable.d.ts
@foo//:node_modules/typescript/lib/lib.esnext.array.d.ts
@foo//:node_modules/typescript/lib/lib.es6.d.ts
@foo//:node_modules/typescript/lib/lib.es5.d.ts
@foo//:node_modules/typescript/lib/lib.es2018.full.d.ts
@foo//:node_modules/typescript/lib/lib.es2018.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.typedarrays.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.string.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.sharedmemory.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.object.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.intl.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.full.d.ts
@foo//:node_modules/typescript/lib/lib.es2017.d.ts
@foo//:node_modules/typescript/lib/lib.es2016.full.d.ts
@foo//:node_modules/typescript/lib/lib.es2016.d.ts
@foo//:node_modules/typescript/lib/lib.es2016.array.include.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.symbol.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.reflect.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.proxy.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.promise.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.iterable.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.generator.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.core.d.ts
@foo//:node_modules/typescript/lib/lib.es2015.collection.d.ts
@foo//:node_modules/typescript/lib/lib.dom.iterable.d.ts
@foo//:node_modules/typescript/lib/lib.dom.d.ts
@foo//:node_modules/typescript/lib/lib.d.ts
@foo//:node_modules/typescript/lib/ko/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/ja/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/it/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/fr/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/es/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/de/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/cs/diagnosticMessages.generated.json
@foo//:node_modules/typescript/lib/cancellationToken.js
@foo//:node_modules/typescript/lib/README.md
@foo//:node_modules/typescript/bin/tsserver
@foo//:node_modules/typescript/bin/tsc
@foo//:node_modules/typescript/ThirdPartyNoticeText.txt
@foo//:node_modules/typescript/README.md
@foo//:node_modules/typescript/LICENSE.txt
@foo//:node_modules/typescript/CopyrightNotice.txt
@foo//:node_modules/typescript/CONTRIBUTING.md
@foo//:node_modules/typescript/AUTHORS.md
@foo//:node_modules/typescript/.parallelperf.json
@foo//:node_modules/typescript/.mailmap
@foo//:node_modules/typescript/.gitmodules
```

**Comment by kolotaev** [comment]
I have the similar problem. Workflow is somewhat like that (simplified):
- tar the contents of `@name_of_the_installed_npm_install_repo//:node_modules`
- un-tar it inside build directory. So we have here the following dirs: `src`, `node_modules` and various files like `package.json`, `angular.json`, etc.
- run `$(location @nodejs//:npm) run build` where build is declared as `ng build` in project's `package.json`.

Result: npm ERR! missing script: build

After investigation it occurred that the contents of `@name_of_the_installed_npm_install_repo//:node_modules` has no `.bin` directory when I dereference it (in my case before tarring).

Worked in: rules_nodejs-0.4.1
Bug observed in: rules_nodejs-0.9.1, rules_nodejs-0.9.1, 0.10.0

**Comment by kolotaev** [comment]
Seems, the bug was introduced by this PR: https://github.com/bazelbuild/rules_nodejs/pull/130

**Comment by gregmagolan** [comment]
Resolved by #242
